src/main/kotlin/org/rust/ide/inspections/RsWrongGenericArgumentsNumberInspection.kt

```diff
@@ -53,8 +61,19 @@ class RsWrongTypeArgumentsNumberInspection : RsLocalInspectionTool() {
             else -> null
         } ?: return
 
-        val problemText = "Wrong number of type arguments: expected ${errorText}, found $actualArgs"
-        val fixes = getFixes(o, actualArgs, expectedTotalParams)
+        val argumentName = when {
+            (expectedTotalTypeParams > 0 || actualTypeArgs > 0) &&
+                expectedTotalConstParams == 0 &&
+                actualConstArgs == 0 -> "type"
+            (expectedTotalConstParams > 0 || actualConstArgs > 0) &&
+                expectedTotalTypeParams == 0 &&
+                actualTypeArgs == 0 -> "const"
+            else -> "generic"
+        }
```

Comment:
nit:
```suggestion
        val haveTypeParams = expectedTotalTypeParams > 0 || actualTypeArgs > 0
        val haveConstParams = expectedTotalConstParams > 0 || actualConstArgs > 0
        val argumentName = when {
            haveTypeParams && !haveConstParams -> "type"
            !haveTypeParams && haveConstParams -> "const"
            else -> "generic"
        }
```

I suppose it's slightly more readable